src/main/java/com/splunk/logtosplunk/event_loggers/BlockEventLogger.java

```diff
@@ -0,0 +1,103 @@
+package com.splunk.logtosplunk.event_loggers;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.splunk.logtosplunk.LogToSplunkMod;
+import com.splunk.logtosplunk.SplunkMessagePreparer;
+import com.splunk.logtosplunk.actions.BlockEventAction;
+import com.splunk.logtosplunk.loggable_events.LoggableBlockEvent;
+
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Vec3;
+import net.minecraft.world.World;
+import net.minecraftforge.event.world.BlockEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+/**
+ * Handles the logging of block events.
+ */
+public class BlockEventLogger {
+    private static final String LOG_NAME_MODIFIER = " - BLOCK";
+    private static final Logger logger = LogManager.getLogger(LogToSplunkMod.LOGGER_NAME + LOG_NAME_MODIFIER);
+
+    /**
+     * Prepares and forwards data to be sent to splunk.
+     */
+    private SplunkMessagePreparer messagePreparer;
+
+    /**
+     * Constructor.
+     *
+     * @param messagePreparer used to process this classes captured data.
+     */
+    public BlockEventLogger(SplunkMessagePreparer messagePreparer) {
```

Comment:
All EventLoggers take a messagePreparer and I can't think of an example where they wouldn't based on the current structure. Does it make sense to pull this out into a superclass?